x-pack/plugins/alerts/server/alert_type_registry.ts

```diff
@@ -96,10 +103,10 @@ export class AlertTypeRegistry {
   }
 
   public register<
-    Params extends AlertTypeParams = AlertTypeParams,
-    State extends AlertTypeState = AlertTypeState,
-    InstanceState extends AlertInstanceState = AlertInstanceState,
-    InstanceContext extends AlertInstanceContext = AlertInstanceContext
+    Params extends AlertTypeParams,
+    State extends AlertTypeState,
+    InstanceState extends AlertInstanceState,
+    InstanceContext extends AlertInstanceContext
```

Comment:
Removed defaults so we force the developers to be explicit about the types they expect.
This should encourage them to use real types rather than just `Record<string, whatever>`